mis_builder/models/mis_builder.py

```diff
@@ -0,0 +1,829 @@
+         'Period name should be unique by report'),
+    ]
+
+    def compute_domain(self, cr, uid, ids, account_, context=None):
+        if isinstance(ids, (int, long)):
+            ids = [ids]
+        domain = []
+        this = self.browse(cr, uid, ids, context=context)[0]
+        env = Environment(cr, uid, {})
+        aep = AccountingExpressionProcessor(env)
+        aep.parse_expr(account_)
+        aep.done_parsing([('company_id', '=',
+                           this.report_instance_id.company_id.id)])
+        domain.extend(aep.get_aml_domain_for_expr(account_))
+        if domain != []:
+            # compute date/period
+            period_ids = []
+            date_from = None
+            date_to = None
+            period_obj = self.pool['account.period']
+            target_move = this.report_instance_id.target_move
+            if target_move == 'posted':
+                domain.append(('move_id.state', '=', target_move))
+            if this.period_from:
+                compute_period_ids = period_obj.build_ctx_periods(
+                    cr, uid, this.period_from.id, this.period_to.id)
+                period_ids.extend(compute_period_ids)
+            else:
+                date_from = this.date_from
+                date_to = this.date_to
+            if period_ids:
+                if date_from:
+                    domain.append('|')
+                domain.append(('period_id', 'in', period_ids))
+            if date_from:
+                domain.extend([('date', '>=', date_from),
+                               ('date', '<=', date_to)])
+        else:
+            domain = False
+        return domain
+
+    def compute_period_domain(self, cr, uid, period_report, is_solde,
+                              is_initial, context=None):
+        period_obj = self.pool['account.period']
+        move_obj = self.pool['account.move']
+        domain_list = []
+        target_move = period_report.report_instance_id.target_move
+        if target_move == 'posted':
+            domain_list.append(('move_id.state', '=', target_move))
+        if not is_solde and not is_initial:
+            if period_report.period_from:
+                compute_period_ids = period_obj.build_ctx_periods(
+                    cr, uid, period_report.period_from.id,
+                    period_report.period_to.id)
+                domain_list.extend([('period_id', 'in', compute_period_ids)])
+            else:
+                domain_list.extend([('date', '>=', period_report.date_from),
+                                    ('date', '<=', period_report.date_to)])
+        elif period_report.period_from and period_report.period_to:
+            period_to = period_report.period_to
+            if is_initial:
+                move_id = move_obj.search(
+                    cr, uid, [('period_id.special', '=', False),
+                              ('period_id.date_start', '<',
+                               period_to.date_start)],
```

Comment:
Ok...

create a new db with demo data.
Install mis_builder_demo.
Install multi_company.
Set chart of account for companies : Your Company and OpenERP BE (admin must be set in group Multi-company)
(for the period of Your Company, do not forget to uncheck special to avoid other error)
Set demo user with the following right : Financial Manager, Settings and Multi-company.
Connect you as demo user, and change company to OpenERP BE
Into Reporting>Configuration>MIS report instance
Create an instance for any Report, with a period of type fiscal period
Click on Preview, all is ok.
Change company of Demo user to set Your Company.
Click on Preview, I get this error message : "Error! You should choose the periods that belong to the same company."


So what should I except ?